File: benches/mutations/variants/python/seed3.t4-mutual-recursion.py
def reduce_pair(a, b, log):
    if b == 0:
        return a
    return shrink_pair(a, b, log)


def shrink_pair(a, b, log):
    log.append("step: " + str(a) + " " + str(b))
    return reduce_pair(b, a % b, log)

File: benches/mutations/variants/rust/seed3.t4-mutual-recursion.rs
fn reduce_pair(a: u64, b: u64, log: &mut Vec<String>) -> u64 {
    if b == 0 {
        return a;
    }
    shrink_pair(a, b, log)
}

fn shrink_pair(a: u64, b: u64, log: &mut Vec<String>) -> u64 {
    log.push(format!("step: {a} {b}"));
    reduce_pair(b, a % b, log)
}
